### PR TITLE
Add functionality overall

### PR DESCRIPTION
**Pull request by GaithAST**
I really dont remember what exactly I did, just the basic image picker. Sorry.

### CODE REVIEW

**Comment by GaithAST** [comment]
Please ignore the config, forgot to .gitignore it

File: pb_app/lib/config.dart [hunk 1]
```diff
@@ -1,7 +1,7 @@
 class Config {
   static const baseUrl = String.fromEnvironment(
     'baseUrl',
-    defaultValue: 'http://127.0.0.1:8090',
+    defaultValue: 'http://192.168.100.8:8090',
```
Comment (lukepighetti) [inline pb_app/lib/config.dart]:
```suggestion
    defaultValue: 'http://127.0.0.1:8090',
```

File: pb_app/lib/modals.dart [hunk 2]
```diff
@@ -32,12 +33,29 @@ class _LoginModalState extends State<LoginModal> {
...
+                color: Colors.white,
+              ),
+              SizedBox(
+                width: 16,
+              ),
+              Text('Signed in successfully!'),
+            ],
+          ),
+        ),
+      );
```
Comment (lukepighetti) [inline pb_app/lib/modals.dart]: I don't think we need to notify the user of success, the navigation to a feature that requires authentication will inform them of the success :)
```suggestion
```

File: pb_app/lib/submission_form.dart [hunk 3]
```diff
@@ -84,8 +86,12 @@ class _SubmissionFormScreenState extends State<SubmissionFormScreen> {
...
       body: PageView(
         children: const [
-          _Card('Home Screen'),
-          _Card('Lock Screen'),
+          _Card(
+            'Home Screen',
+          ),
+          _Card(
+            'Lock Screen',
+          ),
```
Comment (lukepighetti) [inline pb_app/lib/submission_form.dart]:
```suggestion
          _Card('Home Screen'),
          _Card('Lock Screen'),
```

Best to leave out as many style changes as possible :)

File: pb_app/lib/submission_form.dart [hunk 4]
```diff
@@ -104,24 +110,71 @@ class _SubmissionFormScreenState extends State<SubmissionFormScreen> {
   }
 }
 
-class _Card extends StatelessWidget {
-  const _Card(this.title);
-
+class _Card extends StatefulWidget {
+  const _Card(
+    this.title,
+  );
```
Comment (lukepighetti) [inline pb_app/lib/submission_form.dart]:
```suggestion
  const _Card(this.title);
```

revert pure style change

File: pb_app/lib/workflows.dart [hunk 5]
```diff
@@ -2,15 +2,16 @@ import 'package:flutter/widgets.dart';
...
 class Workflows {
-  static Future<UserAuth> authenticate() {
+  static Future<UserAuth> authenticate({email, password}) {
     return client.users.authViaEmail(
-      Secrets.email,
-      Secrets.password,
+      email ?? Secrets.email,
+      password ?? Secrets.password,
     );
   }
```
Comment (lukepighetti) [inline pb_app/lib/workflows.dart]: Ah, nice find. Here's my suggestion: delete this authenticate workflow and create an extension on `client.users` called `authViaSecrets` with no arguments and use that instead

File: pb_app/pubspec.yaml [hunk 6]
```diff
@@ -7,20 +7,20 @@ environment:
...
-  collection:
+  collection: null
   cupertino_icons: ^1.0.2
   figma_squircle: ^0.5.3
+  flutter:
+    sdk: flutter
+  image_picker: ^0.8.5+3
   pocketbase: ^0.4.0
-  provider:
+  provider: null
```
Comment (lukepighetti) [inline pb_app/pubspec.yaml]: careful using plugins to automatically sort these, leave `flutter` at top and sort them manually. I like using `sort descending` tool in VSCode

File: pb_app/pubspec.yaml [hunk 7]
```diff
@@ -7,20 +7,20 @@ environment:
...
 
 dev_dependencies:
-  flutter_test:
-    sdk: flutter
 
   flutter_lints: ^2.0.0
+  flutter_test:
+    sdk: flutter
```
Comment (lukepighetti) [inline pb_app/pubspec.yaml]: same here

**Review by lukepighetti — CHANGES_REQUESTED** [review]
You found some good stuff, but I've requested some changes!

**Comment by lukepighetti** [comment]
Also, let's get this up to date with main branch so we can get ci checks on it

**Comment by GaithAST** [comment]
Fixed the issues! Hope its all good now!

File: pb_app/lib/main.dart [hunk 8]
```diff
@@ -137,3 +139,9 @@ class _MyHomePageState extends State<MyHomePage> {
     );
   }
 }
+
+extension on UserService {
+  get authViaSecrets {
```
Comment (lukepighetti) [inline pb_app/lib/main.dart]: can we type this? Future<void> and also make it a function

File: pb_app/lib/submission_form.dart [hunk 9]
```diff
@@ -104,24 +106,69 @@ class _SubmissionFormScreenState extends State<SubmissionFormScreen> {
...
-
   final String title;
 
+  @override
+  State<_Card> createState() => _CardState();
+}
+
+class _CardState extends State<_Card> {
+  File? image;
+  final errorSnackBar = SnackBar(
```
Comment (lukepighetti) [inline pb_app/lib/submission_form.dart:120]: Please use the built in error toast service :)

File: pb_app/lib/workflows.dart [hunk 10]
```diff
@@ -2,16 +2,17 @@ import 'package:flutter/widgets.dart';
...
-  static Future<UserAuth> authenticate() {
-    return client.users.authViaEmail(
-      Secrets.email,
-      Secrets.password,
-    );
+  static Future<UserAuth> authenticate(email, password) {
+    if (email != null && password != null) {
+      return client.users.authViaEmail(email, password);
+    }
+    return client.users.authViaSecrets;
```
Comment (lukepighetti) [inline pb_app/lib/workflows.dart]: We can remove Workflows.authenticate() method now that we have the authViaSecrets extension :)

File: pb_app/lib/workflows.dart [hunk 11]
```diff
@@ -34,9 +35,13 @@ class Workflows {
...
+      SubmissionFormScreen.show(context);
     }
   }
 }
+
+extension on UserService {
+  get authViaSecrets {
+    return UserService(client).authViaEmail(Secrets.email, Secrets.password);
+  }
+}
```
Comment (lukepighetti) [inline pb_app/lib/workflows.dart]: let's move this into `utils.dart` and remove the duplicate in `main.dart`

File: pb_app/pubspec.yaml [hunk 12]
```diff
@@ -9,17 +9,16 @@ environment:
 dependencies:
   flutter:
     sdk: flutter
-
-  collection:
+  collection: null
```
Comment (lukepighetti) [inline pb_app/pubspec.yaml]: Please remove the `null` values OR change them to `any`

**Review by lukepighetti — CHANGES_REQUESTED** [review]
A couple more things, thank you so much!

File: pb_app/lib/workflows.dart (hunk 10 above)
Comment (GaithAST) [reply pb_app/lib/workflows.dart]: I think we need it for the login modal

Comment (lukepighetti) [reply pb_app/lib/workflows.dart]: in login modal, can we do `authViaEmail`?

Comment (GaithAST) [reply pb_app/lib/workflows.dart]: we can, but I dont see whats wrong with this. 

Ill do it anyways though

**Comment by GaithAST** [comment]
I hope all's good now!

**Comment by lukepighetti** [comment]
Hey @GaithAST, I want to push to your PR branch. Can you enable "Allow edits from maintainers"? https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests

**Comment by lukepighetti** [comment]
merged and closed by #9, thank you!!!

**Pull request closed without merging**